lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -89,31 +85,41 @@ public synchronized void onRestart() {
     final EvaluatorRestartInfo evaluatorRestartInfo = driverRuntimeRestartManager.getAliveAndFailedEvaluators();
     setPreviousEvaluatorIds(evaluatorRestartInfo.getAliveEvaluators());
     driverRuntimeRestartManager.informAboutEvaluatorFailures(evaluatorRestartInfo.getFailedEvaluators());
+    // TODO[REEF-560]: Call onDriverRestartCompleted() (to do in REEF-617) on a Timer.
   }
 
   /**
-   * @return whether restart is completed.
+   * @return Whether restart is completed.
    */
   public synchronized boolean isRestartCompleted() {
     return this.state == DriverRestartState.RestartCompleted;
   }
 
   /**
-   * @return the Evaluators expected to check in from a previous run.
+   * @return The restart state of the specified evaluator. Returns {@link EvaluatorRestartState#NOT_RESTARTED_EVALUATOR}
+   * if the {@link DriverRestartManager} does not believe that it's an evaluator to be recovered.
    */
-  public synchronized Set<String> getPreviousEvaluatorIds() {
-    return Collections.unmodifiableSet(this.previousEvaluators);
+  public synchronized EvaluatorRestartState getEvaluatorRestartState(final String evaluatorId) {
+    if (!this.state.hasRestarted() ||
```

Comment:
It might make sense to code the negation into a method, e.g. `state.hasNotRestarted()`